### PR TITLE
Add common role

### PR DESCRIPTION
**Pull request by mlangbehn**
Common role now:
- Exists at all
- Installs extra packages (vim, and zsh)
- Allows passwordless sudo for sudo group
- Creates system users
- Configures SSH
- Configures UFW
  - Policy: Deny
  - OpenSSH: Limit traffic

Signed-off-by: Matt Langbehn <matthew.langbehn@gmail.com>

### CODE REVIEW

File: site.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Common role for all hosts
+  hosts: all
+  become: true
```
Comment (eggshell) [inline site.yml:4]: don't you need `become_method` or `become_user` here?

File: roles/common/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    state: restarted
+
+- name: Enable OpenSSH in UFW
+  ufw:
+    rule: limit
+    name: OpenSSH
+
+- name: Enable UFW, deny traffic by default, and turn on logging
+  ufw:
+    state: "enabled"
```
Comment (eggshell) [inline roles/common/tasks/main.yml]: enabled doesn't need quotes.

File: roles/common/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    policy: deny
+    logging: "on"
```
Comment (eggshell) [inline roles/common/tasks/main.yml]: logging *probably* doesn't need quotes.

File: roles/common/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+- name: Enable SSH pubkey authentication
+  lineinfile:
+    dest: /etc/ssh/sshd_config
+    regexp: 'PubkeyAuthentication'
+    line: 'PubkeyAuthentication yes'
+
+- name: Restart sshd
+  service:
+    name: "sshd"
```
Comment (eggshell) [inline roles/common/tasks/main.yml]: name and enabled shouldn't need quotes

File: inventories/group_vars/all.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
+extra_packages:
+  - vim
+  - zsh
+
+system_users:
+  - eggshell:
+    name: eggshell
+    groups: sudo
+    shell: /usr/bin/zsh
```
Comment (eggshell) [inline inventories/group_vars/all.yml:10]: You know me so well.

**Review by eggshell — CHANGES_REQUESTED** [review]
Looks great! Few nits inline.

File: site.yml (hunk 1 above)
Comment (mlangbehn) [reply site.yml:4]: Not with ansible 2.x, no. `become` defaults to "use sudo to become root."

File: inventories/group_vars/all.yml (hunk 5 above)
Comment (mlangbehn) [reply inventories/group_vars/all.yml:10]: ;-)

File: roles/common/tasks/main.yml (hunk 3 above)
Comment (mlangbehn) [reply roles/common/tasks/main.yml]: I'll test it both ways.

File: roles/common/tasks/main.yml (hunk 2 above)
Comment (mlangbehn) [reply roles/common/tasks/main.yml]: Cool, quotes removed.

File: roles/common/tasks/main.yml (hunk 4 above)
Comment (mlangbehn) [reply roles/common/tasks/main.yml]: Quotes removed from the name. I'll test `enabled: true`.

File: site.yml (hunk 1 above)
Comment (eggshell) [reply site.yml:4]: word

File: roles/common/tasks/main.yml (hunk 4 above)
Comment (mlangbehn) [reply roles/common/tasks/main.yml]: Tested, it worked.

File: roles/common/tasks/main.yml (hunk 3 above)
Comment (mlangbehn) [reply roles/common/tasks/main.yml]: Tested, it worked.

**Review by eggshell — APPROVED** [review]

**Pull request merged**